k8s/upgrades/0.9.0-1.0.0/jiva/jiva_volume_upgrade.sh

```diff
@@ -0,0 +1,177 @@
+fi
+replica_version=`kubectl get deployment $r_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$replica_version" != "$current_version" ]] && [[ "$replica_version" != "$target_upgrade_version" ]]; then
+    echo "Current Replica deployment $r_dep version is not $current_version or $target_upgrade_version";exit 1;
+fi
+
+controller_svc_version=`kubectl get svc $c_svc -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$controller_svc_version" != $current_version ]] && [[ "$controller_svc_version" != "$target_upgrade_version" ]] ; then
+    echo "Current Target service $c_svc version is not $current_version or $target_upgrade_version";exit 1;
+fi
+
+sed -u "s/@r_name@/$r_name/g" jiva-replica-patch.tpl.json | sed -u "s/@target_version@/$target_upgrade_version/g" | sed -u "s/@pv_name@/$pv/g" > jiva-replica-patch.json
+sed -u "s/@c_name@/$c_name/g" jiva-target-patch.tpl.json | sed -u "s/@target_version@/$target_upgrade_version/g" > jiva-target-patch.json
+sed -u "s/@target_version@/$target_upgrade_version/g" jiva-target-svc-patch.tpl.json > jiva-target-svc-patch.json
+
+#################################################################################
+# STEP: Patch OpenEBS volume deployments (jiva-target, jiva-replica & jiva-svc) #
+#################################################################################
+
+# PATCH JIVA REPLICA DEPLOYMENT ####
+if [[ "$replica_version" != "$target_upgrade_version" ]]; then
+    echo "Upgrading Replica Deployment to $target_upgrade_version"
+
+    kubectl patch deployment --namespace $ns $r_dep -p "$(cat jiva-replica-patch.json)"
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to patch the deployment $r_dep | Exit code: $rc"; exit; fi
+
+    kubectl delete rs $r_rs --namespace $ns
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to delete ReplicaSet $r_rs  | Exit code: $rc"; exit; fi
+
+    rollout_status=$(kubectl rollout status --namespace $ns deployment/$r_dep)
+    rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+    then echo " RollOut for $r_dep failed | Exit code: $rc"; exit; fi
+else
+    echo "Replica Deployment $r_dep is already at $target_upgrade_version"
+fi
+
+# #### PATCH TARGET DEPLOYMENT ####
+if [[ "$controller_version" != "$target_upgrade_version" ]]; then
+    echo "Upgrading Target Deployment to $target_upgrade_version"
+
+    kubectl patch deployment  --namespace $ns $c_dep -p "$(cat jiva-target-patch.json)"
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to patch deployment $c_dep | Exit code: $rc"; exit; fi
+
+    kubectl delete rs $c_rs --namespace $ns
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to patch deployment $c_rs | Exit code: $rc"; exit; fi
+
+    rollout_status=$(kubectl rollout status --namespace $ns  deployment/$c_dep)
+    rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
```

Comment:
rollout status will take some time to give success.
Do we need to have some retries?

Reply:
the rollout status runs in the foreground so retries are not required